src/dagmc/DagMC.cpp

```diff
@@ -0,0 +1,774 @@
+#include "DagMC.hpp"
+
+#include <string>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <limits>
+#include <algorithm>
+#include <set>
+
+#include <ctype.h>
+#include <string.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+#include <math.h>
+#ifndef M_PI  /* windows */
+# define M_PI 3.14159265358979323846
+#endif
+
+#define MB_OBB_TREE_TAG_NAME "OBB_TREE"
+#define FACETING_TOL_TAG_NAME "FACETING_TOL"
+
+namespace moab {
+
+/* Tolerance Summary
+
+   Facet Tolerance:
+   Maximum distance between continuous solid model surface and faceted surface.
+     Performance:  increasing tolerance increased performance (fewer triangles)
+     Robustness:   should not be affected
+     Knowledge:    user must understand how coarser faceting influences accuracy
+                   of results
+*/
+
+const bool counting = false; /* controls counts of ray casts and pt_in_vols */
+
+// Empty synonym map for DagMC::parse_metadata()
+const std::map<std::string, std::string> DagMC::no_synonyms;
+
+// DagMC Constructor
+DagMC::DagMC(Interface* mb_impl, double overlap_tolerance, double p_numerical_precision) {
+  moab_instance_created = false;
+  // if we arent handed a moab instance create one
+  if (NULL == mb_impl) {
+    mb_impl = new moab::Core();
+    moab_instance_created = true;
+  }
+
+  // set the internal moab pointer
+  MBI = mb_impl;
+
+  // make new GeomTopoTool and GeomQueryTool
+  GTT = new moab::GeomTopoTool(MBI, false);
+  GQT = new moab::GeomQueryTool(GTT, overlap_tolerance, p_numerical_precision);
+
+  // This is the correct place to uniquely define default values for the dagmc settings
+  defaultFacetingTolerance = .001;
+}
+
+// Destructor
+DagMC::~DagMC() {
+  // delete the GeomTopoTool and GeomQueryTool
+  delete GTT;
+  delete GQT;
+
+  // if we created the moab instance
+  // clear it
+  if (moab_instance_created) {
+    MBI->delete_mesh();
+    delete MBI;
+  }
+}
+
+// get the float verision of dagmc version string
+float DagMC::version(std::string* version_string) {
+  if (NULL != version_string)
+    *version_string = std::string("DagMC version ") + std::string(DAGMC_VERSION_STRING);
+  return DAGMC_VERSION;
+}
+
+unsigned int DagMC::interface_revision() {
+  unsigned int result = 0;
+  const char* interface_string = DAGMC_INTERFACE_REVISION;
+  if (strlen(interface_string) >= 5) {
```

Comment:
Optional: It may be prudent to use `string::rfind` [1] to check if `$Rev: ` exists in the string.  To be super robust, you could also create a regex [2] to locate `$Rev: ` and explicitly check if it is followed by a number

[1] http://www.cplusplus.com/reference/string/string/rfind/
[2] http://www.cplusplus.com/reference/regex/